### DISCUSSION TITLE
Old Stock or new PCB?

### DISCUSSION THREAD

**Discussion by LimeProgramming** [keirf/flashfloppy#636] [General]
Hello there,

A gotek I ordered arrived recently and finally getting a chance to take the thing apart I was surprised to see it had an STM32 in it, not an Artery chip. It also has the new ports for the stock rotary encoder. I didn't think they existed.

It made installing a rotary encoder quite confusing. Wiring it for the Artery didn't work and it was missing the extra 2 pins beside the pins for the display. I got something working eventually.

Is this a new board design or is it just old stock?

![IMG_20220412_080500020](https://user-images.githubusercontent.com/29736217/162901410-d1c7458f-154b-4738-86e5-90c8c92f993d.jpg)
![IMG_20220412_080516645](https://user-images.githubusercontent.com/29736217/162901426-4e3f16eb-df25-43b3-b1bd-214c49624766.jpg)

**Comment by keirf** [comment]
Old stock. This was the factory prototype for the OLED/rotary factory model just as the pandemic shortages hit. The pins on KC30 header subsequently got remapped to support Artery 32- and 48-pin chips, and that's the only KC30 mapping supported by FF, as you have found out.

I'd assumed this was only ever a prototype board. I'll have to ask the factory how many were actually produced. I expect it's pretty rare!

**Comment by keirf** [comment]
From the factory:
```
I found the record. 
Only about 100pcs products used this PCB SFRKC30 with STM32F105.
It is a little different from 922d, only 6p PIN.
```
Super rare: You might be the only user to try attaching a rotary encoder to one of these!

Luckily JK pin row has the old rotary directional pins I believe. I don't think there's any select (rotary push button) pin that works on that header block. You probably had to connect to JA?

**Comment by LimeProgramming** [comment]
well that's unexpected. NGL I slightly regret modifying this unicorn.

I think I'll order another from the same amazon seller and we'll see what shows up. They have 5 left of a Roland Model, so there's a chance it's all the same batch. 

Yea I connected it to JA from underneath to make the wiring neater.

**Comment by LimeProgramming** [comment]
If anyone is interested

https://www.amazon.co.uk/dp/B07N1B1GTG?psc=1&smid=A2KREUBK79VF64&ref_=chk_typ_imgToDp

that's the link to the one I bought. 4 left

I was keeping an eye out for a while and they appeared for relatively cheap. I assumed it was just a seller tired of of their stock or something. No idea if the same will show up again but who knows.

**Reply by keirf** [reply]
I've bought one. Hopefully it will be the same PCB model. I like at least one of each model.

**Reply by LimeProgramming** [reply]
Let me know if it does. It would be curious if the seller just had a batch of these.

**Reply by keirf** [reply]
I received it and it too is SFRKC30. Programmed via USB. I haven't actually had a 2-digit LED display before. It flickers for a few seconds unless I manually override a setting in FF.CFG. How interesting!

**Reply by LimeProgramming** [reply]
> I received it and it too is SFRKC30. Programmed via USB. I haven't actually had a 2-digit LED display before. It flickers for a few seconds unless I manually override a setting in FF.CFG. How interesting!

Odd, mine didn't flicker, just kept saying "up" thanks to the jumper. Still glad to hear you got the SFRKC30. 
My second one is still in the post, Hopefully it's the same.

**Comment by LimeProgramming** [comment]
Oh, 

For the record, the JK pin row has the old rotary directional pins. 
First I thought the docs mixed up CLK and SW which I sometimes do, Swapping them didn't seem to work. I started flicking the wheel left and right out of frustration and the screen incremented. So finding one working pin I tried the old order and that worked no problem.

It was very confusing which is why I decided to post instead of just assuming some old stock I had never seen before.

**Reply by keirf** [reply]
Yes, JK still exists on other SFRKC30 models, but the factory didn't choose to connect rotary encoders in the traditional FlashFloppy way, instead using the new J8 pins, and using JK for the push button. Of course I have to follow the factory way for new models otherwise the factory-fitted rotary encoders wouldn't work! Whereas for this in-betweeny model the best we can do is use the relocated JK header, plus the JA header, in the traditional way.

**Reply by keirf** [reply]
I'm sure 99%+ of FlashFloppy users would never have worked any of this out and been utterly confused!!

**Reply by keirf** [reply]
Further info: We have to use the traditional wiring because I cannot differentiate this PCB from the old SFRC922 PCBs in firmware. The trick I use to detect the KC30 header is dependent on Artery chips and the modified pin assignments used on the later mass-produced SFRKC30.ATx boards. So, this board just looks like a SFRC922  to me.

**Reply by LimeProgramming** [reply]
I was very confused. 

Really only getting there by fluke. 

Out of the box for this model there is a jumper on JA which resulted in the device starting up saying "up" on the two digit display. 
The OLED gave the update flash floppy message. 


Still fascinating that this previously only thought factory prototype turned up in the hands of someone who got the rotary encoder wired in right. Keeping in mind that at the time of purchase I didn't notice it had a 2 digit display, a model listed to avoid. 
Lots of flukes had to come together for that.

Still saying that I got a limited edition Floppy Disk Drive emulator is the nerdiest brag ever. I'll have to put that into a bio somewhere.

Thanks for the information on this model though. Quite a curious case.

**Comment by LimeProgramming** [comment]
Also, I forgot to say.

For flashing it use the USB Programming way. Trying to use a USB to TTL device did not work, the software didn't recognize it. The software would just hang and give up.

I made a usb a to a cable and that worked fine. I think I backed up the original ROM just in case something went wrong with flash floppy. It had read protect but gave no write protect message.

**Reply by keirf** [reply]
I'll let you know how I get on. If it was read protected (and the factory firmware always is afaik) then you have not dumped it: Removing read protection mass erases the flash memory. Otherwise it would be pretty rubbish read protection!

**Reply by LimeProgramming** [reply]
Oh yeah, that would make sense.

I was too focused on my jank USB cable to really think it through. It was two microUSB's connected to two bridged micro usb break outs. I was just hoping that the globs of flux don't conduct and mess up the GOTEK.

**Comment by LimeProgramming** [comment]
The second one I ordered from the same seller arrived and no, not the same prototype board as the first time. 
This time it's a SFRBA16D with an STM32. So at least it's an STM32.

**Reply by keirf** [reply]
I ordered a second one so I'll see what I get. I font have sfrba16 either so I'll be happy to get one.

**Reply by LimeProgramming** [reply]
The cost of this one thread is beginning to add up. 

Let us know what arrives.

**Reply by keirf** [reply]
It's another SFRKC30 for me :)

**Reply by LimeProgramming** [reply]
Alrighty.

I still find it hard to believe that the first gotek I order just happened to be a SFRKC30 with an stm32. The gotek has been on my list of stuff to buy since 2016-ish, checking in for anything new once in the blue moon and finally bothering to nab one in 2022.
I have to wonder what the odds of that are. 

Unless you're interested in a swap, I guess this story has reached it's end.
It was a fun ride while it lasted.

Take it easy dude!
-Lime

**Reply by keirf** [reply]
Happy to swap for SFRBA16. My second one has one busted segment on the crappy two-digit LED, is the only caveat. Board otherwise works fine, and you probably don't care about the horrible display anyway. If you want to swap just drop me an email, address is on my profile.

**Reply by ejona86** [reply]
I ordered one between keirf's first and second order. I just received it and it is a SFRKC30.

**Reply by LimeProgramming** [reply]
> I ordered one between keirf's first and second order. I just received it and it is a SFRKC30.

I wonder if the sudden orders confused the seller. 

Keirf and I swapped his SFRKC30 for my SFRBA16. So that's something fun for the shelf!
